packages/iris-grid/src/IrisGrid.jsx

```diff
@@ -1371,6 +1374,34 @@ export class IrisGrid extends Component {
     this.grid.forceUpdate();
   }
 
+  freezeColumnByColumnName(columnName) {
+    const { frozenColumns } = this.state;
+    const { model } = this.props;
+    log.debug2('freezing column', columnName);
+
+    let allFrozenColumns = new Set([...frozenColumns, columnName]);
+
+    if (frozenColumns.length === 0) {
```

Comment:
This is not the check we want - we only want to pull from the model layout hints if the frozenColumns are `null`. We need to do that to differentiate between two different cases:
1) User hasn't set frozen columns (value is `null`)
2) User has explicitly unfrozen all columns (value is `[]`)